src/server/game/Loot/LootMgr.cpp

```diff
@@ -1426,14 +1427,24 @@ void LootTemplate::LootGroup::Process(Loot& loot, Player const* player, LootStor
                 uint32 maxcount = uint32(float(item->maxcount) * sWorld->getRate(RATE_DROP_ITEM_REFERENCED_AMOUNT));
                 sScriptMgr->OnAfterRefCount(player, loot, rate, lootMode, const_cast<LootStoreItem*>(item), maxcount, store);
                 for (uint32 loop = 0; loop < maxcount; ++loop) // Ref multiplicator
-                    Referenced->Process(loot, store, lootMode, player, item->groupid);
+                    // This reference needs to be processed further, but it is marked isTopLevel=false so that any groups inside
+                    // the reference are not multiplied by Rate.Drop.Item.GroupAmount
+                    Referenced->Process(loot, store, lootMode, player, item->groupid, false);
             }
         }
         else
         {
             // Plain entries (not a reference, not grouped)
             sScriptMgr->OnBeforeDropAddItem(player, loot, rate, lootMode, const_cast<LootStoreItem*>(item), store);
             loot.AddItem(*item); // Chance is already checked, just add
+
+            // If we still have non-ref runs to do for this group AND this item wasn't a reference,
+            // recursively call this function to produce more items for this group.
+            // However, if this is a quest item we shouldn't multiply this group.
+            if (nonRefIterationsLeft > 1 and not item->needs_quest)
```

Comment:
```suggestion
            if (nonRefIterationsLeft > 1 && nonRefIterationsLeft != item->needs_quest)
```

Reply:
Hey, thanks for reviewing! Can you help me understand why your suggestion is comparing the tracking variable `nonRefIterationsLeft` to the bool `item->needs_quest`?

Reply:
I didnt look at the code. it was just an assumption.

But then I want you to change it to

```cpp
if (nonRefIterationsLeft > 1 && !item->needs_quest)
```

Reply:
we don't use `and not`

Reply:
> I didnt look at the code. it was just an assumption.
> 
> But then I want you to change it to
> 
> ```c++
> if (nonRefIterationsLeft > 1 && !item->needs_quest)
> ```

Got it, code style issue. I'll fix that right now.